lib/cli/src/commands/app/secrets/update.rs

```diff
@@ -0,0 +1,266 @@
+            anyhow::bail!(
+                "No app configuration file found in path {}.",
+                path.display()
+            )
+        }
+
+        if self.non_interactive {
+            anyhow::bail!("No app id given. Provide one as a positional argument.")
+        } else {
+            let id = prompt_app_ident("Enter the name of the app")?;
+            let app = id.resolve(client).await?;
+            Ok(app.id.into_inner())
+        }
+    }
+
+    /// Given a list of secrets, checks if the given secrets already exist for the given app and
+    /// returns a list of secrets that must be upserted.
+    async fn filter_secrets(
+        &self,
+        client: &WasmerClient,
+        app_id: &str,
+        secrets: Vec<Secret>,
+    ) -> anyhow::Result<Vec<Secret>> {
+        let names = secrets.iter().map(|s| &s.name);
+        let app_secrets =
+            wasmer_api::query::get_all_app_secrets_filtered(client, app_id, names).await?;
+        let sset = HashSet::<&str>::from_iter(app_secrets.iter().map(|s| s.name.as_str()));
+
+        let mut ret = vec![];
+
+        for secret in secrets {
+            if !sset.contains(secret.name.as_str()) {
+                if self.non_interactive {
+                    anyhow::bail!("Cannot update secret '{}' in app {app_id} as it does not exist yet. Use the `create` command instead.", secret.name.bold());
+                } else {
+                    eprintln!(
+                        "Secret '{}' does not exist for the selected app.",
+                        secret.name.bold()
+                    );
+                    let theme = ColorfulTheme::default();
+                    let res = dialoguer::Confirm::with_theme(&theme)
+                        .with_prompt("Do you want to create it?")
+                        .interact()?;
+
+                    if !res {
+                        eprintln!("Cannot update secret '{}' as it does not exist yet. Use the `create` command instead.", secret.name.bold());
+                    }
+                }
+            }
+
+            ret.push(secret);
+        }
+
+        Ok(ret)
+    }
+    async fn update(
+        &self,
+        client: &WasmerClient,
+        app_id: &str,
+        secrets: Vec<Secret>,
+    ) -> Result<(), anyhow::Error> {
+        let res = wasmer_api::query::upsert_app_secrets(
+            client,
+            app_id,
+            secrets.iter().map(|s| (s.name.as_str(), s.value.as_str())),
+        )
+        .await?;
+        let res = res.context(
+            "Backend did not return any payload to confirm the successful update of the secret!",
+        )?;
+
+        if !res.success {
+            anyhow::bail!("Secret creation failed!")
+        } else {
+            if !self.quiet {
+                eprintln!("Succesfully updated secret(s):");
+                for secret in secrets {
+                    eprintln!("{}", secret.name.bold());
+                }
```

Comment:
As in `create`, should show a hint that it's not deployed yet and user must re-deploy to apply.

Reply:
As above.